nautobot/core/apps/__init__.py

```diff
@@ -94,43 +106,65 @@ def register_menu_items(tab_list):
+                            inner_item,
+                            inner_item.link,
+                            f"{nav_tab.name} -> {group.name} -> {item.name} -> {inner_item.link}",
+                        )
+                        inner_item_perms = set(perms for perms in inner_item.permissions)
+                        registry_groups[group.name]["items"][item.name]["items"][inner_item.link][
+                            "permissions"
+                        ] = inner_item_perms
+                        item_perms |= inner_item_perms
+
+                    # Add collected permissions to group
+                    registry_groups[group.name]["items"][item.name]["permissions"] = item_perms
+                else:
+                    raise TypeError(f"Expected NavMenuItem or NavMenuGroup, but found {item}")
 
-                # Add sorted items to group registry dict
-                registry_groups[group.name]["items"] = OrderedDict(
-                    sorted(registry_groups[group.name]["items"].items(), key=lambda kv_pair: kv_pair[1]["weight"])
-                )
-                # Add collected permissions to group
-                registry_groups[group.name]["permissions"] = group_perms
-                # Add collected permissions to tab
-                tab_perms |= group_perms
+                group_perms |= item_perms
 
-            # Add sorted groups to tab dict
-            registry["nav_menu"]["tabs"][nav_tab.name]["groups"] = OrderedDict(
-                sorted(registry_groups.items(), key=lambda kv_pair: kv_pair[1]["weight"])
+            # Add sorted items to group registry dict
+            registry_groups[group.name]["items"] = OrderedDict(
+                sorted(registry_groups[group.name]["items"].items(), key=lambda kv_pair: kv_pair[1]["weight"])
             )
-            # Add collected permissions to tab dict
-            registry["nav_menu"]["tabs"][nav_tab.name]["permissions"] |= tab_perms
-        else:
-            raise TypeError(f"Top level objects need to be an instance of NavMenuTab: {nav_tab}")
+            # Add collected permissions to group
+            registry_groups[group.name]["permissions"] = group_perms
+            # Add collected permissions to tab
+            tab_perms |= group_perms
+
+        # Add sorted groups to tab dict
+        registry["nav_menu"][nav_tab.name]["groups"] = OrderedDict(
+            sorted(registry_groups.items(), key=lambda kv_pair: kv_pair[1]["weight"])
+        )
+        # Add collected permissions to tab dict
+        registry["nav_menu"][nav_tab.name]["permissions"] |= tab_perms
 
         # Order all tabs in dict
-        registry["nav_menu"]["tabs"] = OrderedDict(
-            sorted(registry["nav_menu"]["tabs"].items(), key=lambda kv_pair: kv_pair[1]["weight"])
+        registry["nav_menu"] = OrderedDict(
+            sorted(registry["nav_menu"].items(), key=lambda kv_pair: kv_pair[1]["weight"])
```

Comment:
FYI this change here to reduce the nesting by one level is why I changed `extras.registry` to allow the top-level keys of the registry to be mutable.

It might be better in revisiting this to revert that change and simply initialize `registry["nav_menu"]` with the fixed ordered list of keys defined in `MENU_TABS` now.